internal/mock/cloudsql.go

```diff
@@ -63,6 +69,54 @@ func WithPublicIP(addr string) FakeCSQLInstanceOption {
 	}
 }
 
+// WithPrivateIP sets the private IP address to addr.
+func WithPrivateIP(addr string) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.ipAddrs["PRIVATE"] = addr
+	}
+}
+
+// WithCertExpiry sets the server certificate's expiration to t.
+func WithCertExpiry(t time.Time) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.Cert.NotAfter = t
+	}
+}
+
+// WithRegion sets the server's region to the provided value.
+func WithRegion(region string) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.region = region
+	}
+}
+
+// WithFirstGenBackend sets the server backend type to FIRST_GEN.
+func WithFirstGenBackend() FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.backendType = "FIRST_GEN"
+	}
+}
```

Comment:
nit: I don't know if we really care about this, there are no first gen instances any more

Reply:
Would you be comfortable removing these lines then?

https://github.com/kurtisvg/cloud-sql-go-connector/blob/cab162906d2bec0862d86a76af1bff2d42d3b2f3/internal/cloudsql/refresh.go#L59-L61